docs/design/proposals/carm/cross-account-resource-management.md

```diff
@@ -0,0 +1,177 @@
+
+## Proposed implementation
+
+#### Overview
+
+The proposed solution will solve the problem of cross account management within a single kubernetes cluster. Given an AWS Owner account ID it will look up an IAM Role ARN and call **STS::AssumeRole** to pivot the AWS SDK calls in order to execute in a new AWS Account and role context.
+
+Controllers will store AccountIDs and their associated IAM Role ARNs in a `ConfigMap` Kubernetes object. Cluster administrators will be responsible for administrating and updating the `ConfigMap`, and creating the target roles using the AWS Console/CLI. Controllers can reuse and existing IAM Role ARN if it matches the scope they need.
+
+To prevent unauthorized users from making any CRUD operations on resources in different AWS accounts, ACK will associate a unique Namespace to each AccountID, which will help administrators limit access to namespaces using K8s RBAC model, hence limit access to AccountIDs.
+
+#### Design details
+
+##### Namespace annotations
+
+To associate a namespace with a unique AWS account, ACK service controllers expect from cluster administrators to create namespaces with a specific annotation `services.k8s.aws/owner-account-id` e.g:
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: production-n1
+  annotations:
+    services.k8s.aws/default-aws-region: "eu-west-1"
+    services.k8s.aws/owner-account-id: "123456789012"
+```
+
+Each time a new AWS resource object is updated, ACK service controllers will lookup the namespaces annotations and decide in which namespace the AWS object should be created.
+
+In addition to the owner account id annotation, users can also specify a `services.k8s.aws/default-aws-region` annotation. This region will be used by default in case region annotation is missing in the CRDs specs.
+
+*//TODO(a-hilaly) address namespace/account-id conflict cases*
+
+*//TODO(a-hilaly) address namespace deletion case*
+
+##### Region determination
+
+To determine within which region the resources should be created, ACK controllers will, in order, look for a region in the following sources: 
+
+- Resource region annotation `services.k8s.aws/aws-region`. If provided it will override the namespace default region annotation.
+- Namespace default region annotation `services.k8s.aws/default-aws-region`
+- If none of the two annotations are provided ACK will try to find a region from this sources: (see [cli user guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-precedence)): Controller flags, Pod IRSA environment variables...
+
+##### Storing AWS Role ARNs
+
+AccountIDs and their associate AWS Role ARNs will be stored in a `ConfigMap` called "ack-carm-config-map". e.g:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ack-carm-config-map
+  namespace: default
+data:
+  accounts: |
+    123456789012=arn:aws:iam::123456789012:root
+    454545454545=arn:aws:iam::454545454545:role/S3Access
```

Comment:
Recommend using the "property-type" keys in the `ConfigMap` YAML instead of the "file-type" keys:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: ack-carm-config-map
  namespace: default
data:
  "123456789012": arn:aws:iam::123456789012:root
  "454545454545": arn:aws:iam::454545454545:role/S3Access
```